modules/navier_stokes/src/fvbcs/INSFVFluxBC.C

```diff
@@ -25,6 +25,71 @@ INSFVFluxBC::INSFVFluxBC(const InputParameters & params)
 {
 }
 
+void
+INSFVFluxBC::computeResidual(const FaceInfo & fi)
+{
+  if (_rc_uo.segregated())
+  {
+    _face_info = &fi;
+    _normal = fi.normal();
+    _face_type = fi.faceType(_var.name());
+
+    if (_face_type == FaceInfo::VarFaceNeighbors::NEIGHBOR)
+      _normal = -_normal;
+
+    if (_face_type == FaceInfo::VarFaceNeighbors::BOTH)
+      mooseError("A FVFluxBC is being triggered on an internal face with centroid: ",
+                 fi.faceCentroid());
+    else if (_face_type == FaceInfo::VarFaceNeighbors::NEITHER)
+      mooseError("A FVFluxBC is being triggered on a face which does not connect to a block ",
+                 "with the relevant finite volume variable. Its centroid: ",
+                 fi.faceCentroid());
+
+    auto r =
+        MetaPhysicL::raw_value(fi.faceArea() * fi.faceCoord() * computeSegregatedContribution());
+
+    if (_face_type == FaceInfo::VarFaceNeighbors::ELEM)
+      prepareVectorTag(_assembly, _var.number());
+    else if (_face_type == FaceInfo::VarFaceNeighbors::NEIGHBOR)
+      prepareVectorTagNeighbor(_assembly, _var.number());
+
+    _local_re(0) = r;
+    accumulateTaggedLocalResidual();
```

Comment:
Can't you just call the base class implementation and remove the code duplication?

Reply:
Ah I see that you are calling `computeSegregatedContribution` instead of `computeQpResidual`. We may have even talked about this, but can you make `computeQpResidual` non-final and then `override` that instead of `computeSegregatedContribution`? Then you could call down to the base-class `computeXXX` when `_rc_uo.segregated()`

Reply:
I had the exact same implementation before the previous review and @GiudGiud suggested not using the qp version. Also, the segregated part might be different, can use older solutions and so on. I would not like to cause confusion if anybody ready the code.

Reply:
Then I would do something like this:
```c++
ADReal
computeQpResidual()
{
  if (_rc_uo.segregated())
    computeSegregatedContribution();
  else
    mooseError("don't call me");
}
```
and then you don't have to re-implement all the code in `FVFluxBC::computeXXX`. Because someone who updates `FVFluxBC` to provide some new feature or bug-fix may not think to come to `INSFVFluxBC` and do the same thing there.

Reply:
Sounds good to me!

Reply:
I don't think the computeQpResidual-based solution works. The thing is, `INSFVFluxBC` is overridden no matter what.

Right now it is overriden to make sure we don't use it in conjunction with gatherRC. 

All I am doing is just adding a special case for segregated solvers.

Reply:
The only way I can think about this is passing a function poointer to a general function. 
So something like `computeXYInternal(function_pointer)` in `FVFlux` and then it is used there and in `INSFVFlux`. Or a lambda.

Reply:
Derived classes of `INSFVFluxBC` should not be overriding `computeResidual`, `computeJacobian`, etc. That's the duplication I am complaining about. I don't think we should be overriding `computeResidual`, `computeJacobian` or `computeResidualAndJacobian` in `INSFVFluxBC` or its derived classes.

So if the complaint is about having derived classes override `computeQpResidual` which doesn't have meaning to a FV developer, then that is why I proposed this implementation of `computeQpResidual` in `INSFVFluxBC`:
```c++
ADReal
INSFVFluxBC::computeQpResidual()
{
  if (_rc_uo.segregated())
    computeSegregatedContribution();
  else
    mooseError("don't call me");
}
```

Reply:
My proposed `computeQpResidual` override would allow us to
1. continue to use the `FVFluxBC` implementations of `computeResidual`, `computeJacobian`, etc.
2. make derived classes override a more intuitive method name `computeSegregatedContribution`

Reply:
Okay, I have a question then. We have the NSMomentumResidualObjects which will call `gatherRCData` and the corresponding kernels and BCs (INSFVFluxKernel and BC) disable the `computeXY` functions. So right now `computeXY` is overridden with -do-nothing-. If I interpret this correctly, even if we override `computeQpResidual`, we will do nothing (I might be missing some inheritance knowledge here).

If we want to disable the current -do nothing- strategy, we will add a little extra computational cost to the monolithic approach because the regular flux loops will pick up either the `return 0.0`s or the errors. I am probably being dense here but are you suggesting I rewrite the loops too to consider segregatedness?

Reply:
Sorry. What I wrote was not quite correct. I forgot that we were already overriding those methods. What I am thinking is in https://github.com/grmnptr/moose/pull/3

Reply:
Okay, added, will resolve this convo.